<ui version="4.0" >
 <class>EFXEditor</class>
 <widget class="QDialog" name="EFXEditor" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>429</width>
    <height>469</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>EFX Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" >
   <item>
    <widget class="QTabWidget" name="m_tab" >
     <property name="toolTip" >
      <string/>
     </property>
     <property name="currentIndex" >
      <number>0</number>
     </property>
     <widget class="QWidget" name="General" >
      <attribute name="title" >
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2" >
       <item row="0" column="0" >
        <widget class="QLabel" name="m_nameLabel" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Preferred" hsizetype="Maximum" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text" >
          <string>EFX name</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="3" >
        <widget class="QLineEdit" name="m_nameEdit" >
         <property name="toolTip" >
          <string>The name of the function being edited</string>
         </property>
        </widget>
       </item>
       <item rowspan="5" row="1" column="0" colspan="3" >
        <widget class="QTreeWidget" name="m_tree" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
           <horstretch>0</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="alternatingRowColors" >
          <bool>true</bool>
         </property>
         <property name="rootIsDecorated" >
          <bool>false</bool>
         </property>
         <property name="itemsExpandable" >
          <bool>false</bool>
         </property>
         <property name="allColumnsShowFocus" >
          <bool>true</bool>
         </property>
         <column>
          <property name="text" >
           <string>Step</string>
          </property>
         </column>
         <column>
          <property name="text" >
           <string>Fixture</string>
          </property>
         </column>
         <column>
          <property name="text" >
           <string>Reverse</string>
          </property>
         </column>
        </widget>
       </item>
       <item row="1" column="3" >
        <widget class="QToolButton" name="m_addFixtureButton" >
         <property name="text" >
          <string>...</string>
         </property>
         <property name="icon" >
          <iconset resource="main.qrc" >
           <normaloff>:/edit_add.png</normaloff>:/edit_add.png</iconset>
         </property>
         <property name="iconSize" >
          <size>
           <width>26</width>
           <height>26</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="2" column="3" >
        <widget class="QToolButton" name="m_removeFixtureButton" >
         <property name="text" >
          <string>...</string>
         </property>
         <property name="icon" >
          <iconset resource="main.qrc" >
           <normaloff>:/edit_remove.png</normaloff>:/edit_remove.png</iconset>
         </property>
         <property name="iconSize" >
          <size>
           <width>26</width>
           <height>26</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="3" column="3" >
        <widget class="QToolButton" name="m_raiseFixtureButton" >
         <property name="text" >
          <string>...</string>
         </property>
         <property name="icon" >
          <iconset resource="main.qrc" >
           <normaloff>:/up.png</normaloff>:/up.png</iconset>
         </property>
         <property name="iconSize" >
          <size>
           <width>26</width>
           <height>26</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="4" column="3" >
        <widget class="QToolButton" name="m_lowerFixtureButton" >
         <property name="text" >
          <string>...</string>
         </property>
         <property name="icon" >
          <iconset resource="main.qrc" >
           <normaloff>:/down.png</normaloff>:/down.png</iconset>
         </property>
         <property name="iconSize" >
          <size>
           <width>26</width>
           <height>26</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="5" column="3" >
        <spacer name="verticalSpacer" >
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="0" colspan="2" >
        <widget class="QGroupBox" name="m_orderGroup" >
         <property name="title" >
          <string>Fixture order</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5" >
          <item>
           <widget class="QRadioButton" name="m_parallelRadio" >
            <property name="toolTip" >
             <string>All fixtures move in parallel</string>
            </property>
            <property name="text" >
             <string>Parallel</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_serialRadio" >
            <property name="toolTip" >
             <string>The pattern propagates to each fixture in a sequential order</string>
            </property>
            <property name="text" >
             <string>Serial</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="6" column="2" colspan="2" >
        <widget class="QGroupBox" name="m_busGroup" >
         <property name="title" >
          <string>Speed bus</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6" >
          <item>
           <widget class="QComboBox" name="m_busCombo" >
            <property name="toolTip" >
             <string>Bus that controls the speed of the pattern</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_4" >
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="Movement" >
      <attribute name="title" >
       <string>Movement</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3" >
       <item rowspan="2" row="0" column="0" colspan="2" >
        <widget class="QFrame" name="m_previewFrame" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Fixed" hsizetype="Fixed" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize" >
          <size>
           <width>255</width>
           <height>255</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>255</width>
           <height>255</height>
          </size>
         </property>
         <property name="frameShape" >
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="2" >
        <widget class="QGroupBox" name="m_algorithmGroup" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Maximum" hsizetype="Preferred" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title" >
          <string>Pattern</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3" >
          <item>
           <widget class="QComboBox" name="m_algorithmCombo" >
            <property name="toolTip" >
             <string>Pattern for moving the mirror/head</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_5" >
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Maximum</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item rowspan="2" row="1" column="2" >
        <widget class="QGroupBox" name="m_parametersGroup" >
         <property name="title" >
          <string>Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout" >
          <item row="0" column="0" >
           <widget class="QLabel" name="m_widthLabel" >
            <property name="text" >
             <string>Width</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="QSpinBox" name="m_widthSpin" >
            <property name="toolTip" >
             <string>Value width of the pattern</string>
            </property>
            <property name="maximum" >
             <number>127</number>
            </property>
            <property name="value" >
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" >
           <widget class="QLabel" name="m_heightLabel" >
            <property name="text" >
             <string>Height</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" >
           <widget class="QSpinBox" name="m_heightSpin" >
            <property name="toolTip" >
             <string>Value height of the pattern</string>
            </property>
            <property name="maximum" >
             <number>127</number>
            </property>
            <property name="value" >
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0" >
           <widget class="QLabel" name="m_xOffsetLabel" >
            <property name="text" >
             <string>X offset</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" >
           <widget class="QSpinBox" name="m_xOffsetSpin" >
            <property name="toolTip" >
             <string>Pattern's center point on the X axis</string>
            </property>
            <property name="maximum" >
             <number>255</number>
            </property>
            <property name="value" >
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0" >
           <widget class="QLabel" name="m_yOffsetLabel" >
            <property name="text" >
             <string>Y offset</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" >
           <widget class="QSpinBox" name="m_yOffsetSpin" >
            <property name="toolTip" >
             <string>Pattern's center point on the Y axis</string>
            </property>
            <property name="maximum" >
             <number>255</number>
            </property>
            <property name="value" >
             <number>127</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0" >
           <widget class="QLabel" name="m_rotationLabel" >
            <property name="text" >
             <string>Rotation</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1" >
           <widget class="QSpinBox" name="m_rotationSpin" >
            <property name="toolTip" >
             <string>Rotation of the pattern's starting point</string>
            </property>
            <property name="maximum" >
             <number>359</number>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>20</width>
              <height>16</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="6" column="0" >
           <widget class="QLabel" name="m_xFrequencyLabel" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>X frequency</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1" >
           <widget class="QSpinBox" name="m_xFrequencySpin" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="toolTip" >
             <string>Lissajous pattern's X frequency</string>
            </property>
            <property name="maximum" >
             <number>5</number>
            </property>
            <property name="value" >
             <number>2</number>
            </property>
           </widget>
          </item>
          <item row="7" column="0" >
           <widget class="QLabel" name="m_yFrequencyLabel" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>Y frequency</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1" >
           <widget class="QSpinBox" name="m_yFrequencySpin" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="toolTip" >
             <string>Lissajous pattern's Y frequency</string>
            </property>
            <property name="maximum" >
             <number>5</number>
            </property>
            <property name="value" >
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="8" column="0" >
           <widget class="QLabel" name="m_xPhaseLabel" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>X phase</string>
            </property>
           </widget>
          </item>
          <item row="8" column="1" >
           <widget class="QSpinBox" name="m_xPhaseSpin" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="toolTip" >
             <string>Lissajous pattern's X phase</string>
            </property>
            <property name="maximum" >
             <number>360</number>
            </property>
            <property name="value" >
             <number>90</number>
            </property>
           </widget>
          </item>
          <item row="9" column="0" >
           <widget class="QLabel" name="m_yPhaseLabel" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>Y phase</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1" >
           <widget class="QSpinBox" name="m_yPhaseSpin" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="toolTip" >
             <string>Lissajous pattern's Y phase</string>
            </property>
            <property name="maximum" >
             <number>360</number>
            </property>
            <property name="value" >
             <number>90</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0" >
        <widget class="QGroupBox" name="m_directionGroup" >
         <property name="title" >
          <string>Direction</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2" >
          <item>
           <widget class="QRadioButton" name="m_forward" >
            <property name="toolTip" >
             <string>Run the pattern forwards</string>
            </property>
            <property name="text" >
             <string>Forward</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_backward" >
            <property name="toolTip" >
             <string>Run the pattern backwards</string>
            </property>
            <property name="text" >
             <string>Backward</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2" >
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>20</width>
              <height>38</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="1" >
        <widget class="QGroupBox" name="m_runOrderGroup" >
         <property name="title" >
          <string>Run order</string>
         </property>
         <layout class="QVBoxLayout" name="_2" >
          <item>
           <widget class="QRadioButton" name="m_loop" >
            <property name="toolTip" >
             <string>Run through over and over again</string>
            </property>
            <property name="text" >
             <string>Loop</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_singleShot" >
            <property name="toolTip" >
             <string>Run through once and stop</string>
            </property>
            <property name="text" >
             <string>Single shot</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="m_pingPong" >
            <property name="toolTip" >
             <string>First run forwards, then backwards, again forwards, etc...</string>
            </property>
            <property name="text" >
             <string>Ping pong</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3" >
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0" >
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="Initialization" >
      <attribute name="title" >
       <string>Initialization</string>
      </attribute>
      <layout class="QGridLayout" >
       <item row="0" column="0" >
        <widget class="QGroupBox" name="m_startSceneGroup" >
         <property name="toolTip" >
          <string>Initialize the fixture with the given scene values when the EFX starts</string>
         </property>
         <property name="title" >
          <string>Initialize fixtures</string>
         </property>
         <property name="checkable" >
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" >
          <item>
           <widget class="QTreeWidget" name="m_startSceneList" >
            <property name="alternatingRowColors" >
             <bool>true</bool>
            </property>
            <property name="rootIsDecorated" >
             <bool>false</bool>
            </property>
            <property name="itemsExpandable" >
             <bool>false</bool>
            </property>
            <property name="sortingEnabled" >
             <bool>true</bool>
            </property>
            <property name="allColumnsShowFocus" >
             <bool>true</bool>
            </property>
            <column>
             <property name="text" >
              <string>Scene</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1" >
        <widget class="QGroupBox" name="m_stopSceneGroup" >
         <property name="toolTip" >
          <string>De-initialize the fixture with the given scene values when the EFX stops</string>
         </property>
         <property name="title" >
          <string>De-initialize fixtures</string>
         </property>
         <property name="checkable" >
          <bool>true</bool>
         </property>
         <layout class="QVBoxLayout" >
          <item>
           <widget class="QTreeWidget" name="m_stopSceneList" >
            <property name="alternatingRowColors" >
             <bool>true</bool>
            </property>
            <property name="rootIsDecorated" >
             <bool>false</bool>
            </property>
            <property name="itemsExpandable" >
             <bool>false</bool>
            </property>
            <property name="sortingEnabled" >
             <bool>true</bool>
            </property>
            <property name="allColumnsShowFocus" >
             <bool>true</bool>
            </property>
            <column>
             <property name="text" >
              <string>Scene</string>
             </property>
            </column>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox" >
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons" >
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="main.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EFXEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel" >
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EFXEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel" >
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
